app/topics_handler.go

```diff
@@ -95,18 +95,30 @@ func (th *TopicsHandler) requestTopicReputers(ctx sdk.Context, topic emissionsty
 	} else {
 		maxRetriesToFulfilNoncesReputer = emissionsParams.MaxRetriesToFulfilNoncesReputer
 	}
-	topNReputerNonces := synth.SelectTopNReputerNonces(&reputerNonces, int(maxRetriesToFulfilNoncesReputer), currentBlockHeight, topic.GroundTruthLag)
+	topNReputerNonces := synth.SelectTopNReputerNonces(&reputerNonces, int(maxRetriesToFulfilNoncesReputer), currentBlockHeight, topic.GroundTruthLag, topic.EpochLength)
 	Logger(ctx).Warn(fmt.Sprintf("Iterating Top N Reputer Nonces: %v", len(topNReputerNonces)))
+
+	lastCommit, err := th.emissionsKeeper.GetTopicLastCommit(ctx, topic.Id, emissionstypes.ActorType_REPUTER)
+	if err != nil {
+		Logger(ctx).Warn("Error getting reputer last commit: "+err.Error(), ", first reputer commit?")
+	}
 	// iterate over all the reputer nonces to find if this is unfulfilled
 	for _, nonce := range topNReputerNonces {
 		nonceCopy := nonce
-		Logger(ctx).Warn(fmt.Sprintf("Reputer block height found unfulfilled, requesting reputers for block: %v, worker: %v", nonceCopy.ReputerNonce.BlockHeight, nonceCopy.WorkerNonce.BlockHeight))
+		// Get previous losses from keeper, default: reputer - epochLength
+		lastReputerCommitBlockHeight := nonceCopy.ReputerNonce.BlockHeight - topic.EpochLength
```

Comment:
I don't get why, if `lastCommit.Nonce.BlockHeight` does not exist (as checked below), why anything should exist at this value (nonce minus epoch length)...

Reply:
A tentative value is given, which is the most common value - but not having them should not break the flow, request still needs to be made even if there are no losses at that block at the time of check.
An alternative would be sending a value that marks this explicitly, like 0. Although sending the most common value allows for a last-minute regular reputation being inserted in the in-between time, especially for very short epochLengths.

Reply:
Actually it should be 
`nonceCopy.ReputerNonce.BlockHeight + topic.GroundTruthLag - topic.EpochLength`